scripts/make_statistics.py

```diff
@@ -0,0 +1,539 @@
+# -*- coding: utf-8 -*-
+# SPDX-FileCopyrightText: : 2022 PyPSA-Earth Authors
+#
+# coding: utf-8
+"""
+Execute a scenario optimization
+
+This script contains utility functions to iteratively run the workflow for several regions
+and test the execution for each of them.
+During the execution, statistics on the outputs of the execution are generated.
+Examples of relevant statistics are:
+- For clean_osm_data and download_osm_data,
+  the number of elements, length of the lines and length of dc lines are stored
+- For build_renewable_profiles, total available potential and average production are collected
+- For network rules (base_network, add_electricity, simplify_network and solve_network),
+  length of lines, number of buses and total installed capacity by generation technology
+- Execution time for the rules, when benchmark is available
+
+Outputs
+------
+This rule creates a dataframe containing in the columns the relevant statistics for the current run.
+"""
+import os
+import shutil
+from datetime import datetime
+from pathlib import Path
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import pypsa
+import xarray as xr
+from _helpers import mock_snakemake, sets_path_to_root, to_csv_nafix
+from build_test_configs import create_test_config
+from ruamel.yaml import YAML
+from shapely.validation import make_valid
+
+
+def _multi_index_scen(rulename, keys):
+    return pd.MultiIndex.from_product([[rulename], keys], names=["rule", "key"])
+
+
+def _mock_snakemake(rule, **kwargs):
+    os.chdir(os.path.dirname(os.path.abspath(__file__)))
+    snakemake = mock_snakemake(rule, **kwargs)
+    sets_path_to_root("pypsa-earth")
```

Comment:
Funny that we still use this :D 
Happy to keep it.

Reply:
It could be moved to the helper but it implies changing all files; I'd keep it like this for now